File: algorithms/58.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
    int lengthOfLastWord(string s)
    {
        int length = 0;
        int i = s.size()-1;
        while ( i >= 0 && s[i] == ' ' )
            i--;
        while ( i >= 0 && s[i] != ' ' )
        {
            length++;
            i--;
        }
        return length;
    }
};

int main()
{
    Solution solution;
    cout << solution.lengthOfLastWord("World  ") << endl;
    return 0;
}

File: algorithms/rust/58.rs
impl Solution {
    #[allow(unused)]
    pub fn length_of_last_word(s: String) -> i32 {
        s.split(' ')
            .filter(|word| !word.is_empty())
            .next_back()
            .unwrap()
            .len() as i32
    }
}

#[allow(dead_code)]
struct Solution;

fn main() {
    println!("Hello, world!");
}
